dd-sdk-android/src/main/kotlin/com/datadog/android/rum/GlobalRum.kt

```diff
@@ -153,9 +157,29 @@ object GlobalRum {
 
     internal fun updateRumContext(newContext: RumContext) {
         activeContext.set(newContext)
+        updateContextInPlugins(newContext, RumFeature.pluginsWithRumNotifications)
+        updateContextInPlugins(newContext, CrashReportsFeature.pluginsWithRumNotifications)
+        updateContextInPlugins(newContext, LogsFeature.pluginsWithRumNotifications)
+        updateContextInPlugins(newContext, TracesFeature.pluginsWithRumNotifications)
```

Comment:
This whole thing seems like a very error prone design, and not scalable.
What if we need a to have plugins with APM or other kinds of contexts ? 
I'd be much more comfortable with a more generic solution. 

For example, we could have a global `DatadogContext` object with RUM information (and more as the product grows) and let all DatadogPlugin instances get notified when the global context changes.

Reply:
I agree it was a bit forced, very error prone or not scalable not so sure because that is the reason why I provided the abstraction `PluginWithRumNotifications` to be able to provide more like this if needed (`PluginWithApmNotifications`, etc). Just wanted to have updates only for the context was interested in. I don't like that filter either  I'd rather avoid doing that and have a single DatadogContext object with all the info if we both agree this will not affect perfs (meaning that we could get irrelevant notifications from different features for ex: I am interested in the RumContext but the ApmContext changed and I got a notification to update even though nothing in the RumContext changed).

Reply:
The problem is, how will you create a `PluginWithBothRUMAndAPMNotifications` ? You can't inherit from both, meaning you'll need to create all combination classes, which will grow exponentially with each new feature.
We can be smart about the `DatadogContext` and only fill the information that changed instead and let the implementation of the plugin deal with it (eg: `if (datadogContext.rum != null) …`)